### DISCUSSION TITLE
i beginer i can't sent my notes to github

### DISCUSSION THREAD

**Discussion by dev-hanzo** [ObsidianPublisher/obsidian-github-publisher#63] [Q&A — Plugin] [answered]
**i found two critical mistake**

1. in file "netlify.toml" this strings 
```
[[plugins]]
  package = "netlify-plugin-cache"
  [plugins.inputs]
  paths = [".cache"]
```

they not let netlify maked site 

```Error message
Plugins must be installed either in the Netlify App or in "package.json".
Please run "npm install -D netlify-plugin-cache" or "yarn add -D netlify-plugin-cache"
```

2. Last time i used gigital garden plugin. i tryed all options setting
obsidian-mkdocs-publisher but i can't sent my notes.
publisher plugin don't send my notes to github. 
i use version 4.10.2
i fix first mistake via delete this strings. site started assembled and work

**Comment by Mara-Li** [comment]
Your note are send or not?

**Reply by dev-hanzo** [reply]
i see notice my notes sent, but they not send. **I not see they in my github**



<img width="356" alt="pic 5" src="https://user-images.githubusercontent.com/117753674/210535308-321972cb-354b-41ca-8ca9-1f69bbd32b70.png">

**Comment by Mara-Li** [comment]
What do you have in your console? You can open it using CTRL + MAJ + I

**Comment by Mara-Li** [comment]
Could you check if the repository is found? Mainly around your token and the settings.
I see that the branch is created with a 404 error.

**Reply by dev-hanzo** [reply]
yes, I checked the settings. I made the same settings with the digital garden. Digital garden sends notes with these settings. I will check again.

**Reply by Mara-Li** [reply]
Strange. I dunno how to repro the issue :/ I can just make a button to check the settings of the repo.

**Reply by Mara-Li** [reply]
Can you click on the line indicated by the plugin in the error log?

**Comment by dev-hanzo** [comment]
<img width="490" alt="pic 6" src="https://user-images.githubusercontent.com/117753674/210538477-1353d528-416b-43a6-ad9a-db7f0396b7c2.png">

**settings are the same**

**Reply by dev-hanzo** [reply]
yes, I can click, but I don't know which one exactly. there are a lot of red lines

**Reply by Mara-Li** [reply]
You need to set your main branch.

**Reply by Mara-Li** [reply]
Normally it's main or master

**Reply by dev-hanzo** [reply]
yes, i tried that. It does not help. I did thet again now

**Reply by dev-hanzo** [reply]
![image](https://user-images.githubusercontent.com/117753674/210540506-260e85e2-f58c-40ef-aa56-b8d2a155d924.png)

i see a long line that says everything has been sent

**Reply by dev-hanzo** [reply]
but, in my repo not have my notes

**Reply by dev-hanzo** [reply]
[obsidian.md-1672830444288.log](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10343499/obsidian.md-1672830444288.log)

**i made new log file, check please**

**Reply by Mara-Li** [reply]
Again, the error is in creating branch. Did you set the main branch on main as I said? 

Could you please send me screenshot instead of log file?

**Reply by Mara-Li** [reply]
Also the message bug because of the crash during send. It try to send multiple time the 3 files marked. 
This message doesn't mean the file are send; it display only the number of shared files

**Reply by dev-hanzo** [reply]
I see in log is name repo "publish-mkdocs", but this deprecated repo that I wanted to use. this repo lost. can i made new name? I changed that in setting, but I not connect

**Reply by Mara-Li** [reply]
I don't understand what do you saying...

**Reply by dev-hanzo** [reply]
![image](https://user-images.githubusercontent.com/117753674/210544473-48c25efd-68d4-407e-80ba-93ec57e6ddc3.png)

**Reply by Mara-Li** [reply]
Yeah, the repo seem to have a problem. Is your? On the main branch?

**Reply by dev-hanzo** [reply]
![image](https://user-images.githubusercontent.com/117753674/210545279-9eb920cd-2d62-44e4-8b7c-90b3ab4c09d6.png)

**Reply by dev-hanzo** [reply]
I changed the name repo that in setting, but plugin sent my notes in older repo. maybe I not right

**Reply by dev-hanzo** [reply]
I be back in three hours and I try again

**Reply by Mara-Li** [reply]
I pushed the 4.11.0 version that add a command to test your repository

**Reply by dev-hanzo** [reply]
I tried two ways, but nothing changed.

**Comment by dev-hanzo** [comment]
1. I updated plugin to version 4.11. I had seen this notices, but  my notes did not into github

![image](https://user-images.githubusercontent.com/117753674/210585709-c0332f55-d98a-4feb-bfcf-6ce7be4fae2c.png)

**Comment by dev-hanzo** [comment]
2. I deleted puplisher plugin. I made new repo "pub-2" and new token. I installed  puplisher plugin again and set it up. Nothing changed, my notes did not into repo "pub-2"


![image](https://user-images.githubusercontent.com/117753674/210580372-17cc5a33-517b-4fbd-b108-eff86c79964f.png)

**Reply by Mara-Li** [reply]
Please use the command to check the validity of your repository.

The repo is, here, not found. You don't configure it

**Comment by dev-hanzo** [comment]
![image](https://user-images.githubusercontent.com/117753674/210630271-33ce4810-c700-4c95-aaff-e67449618b2a.png)


![image](https://user-images.githubusercontent.com/117753674/210630459-ad22d391-7c43-4c0f-9e8d-dc8cd0431897.png)

**Reply by Mara-Li** [reply]
Okay, so. It's. Strange.
What is the file you try to send ? 

Try to run the validity check on the file you try to send.

**Reply by dev-hanzo** [reply]
I not undestend "the validity check on the file you try to send."
My three files send by digital garden plugin, but not sending by publisher plugin.

**Reply by Mara-Li** [reply]
The Digital Garden plugin is not Publisher. This is not relevant ; 
I added a commands on each file that can check the validity of your repository. 

Did you put the key `repo` or `multiRepo` in any of your file ?

**Reply by dev-hanzo** [reply]
Did you put the key repo or multiRepo in any of your file ?
**no, I didn't do it. I must do it?**

**Reply by dev-hanzo** [reply]
v. 4.11.1 worke!
 I sent my files to git, but there are two problems 😪😪
1. only png files were sent
2. when I changed in the settings name repo "pub-2"  to name repo "mk-publish" the any files are not sent to another repo with the name "mk-publish"

**Reply by Mara-Li** [reply]
Sorry, I dunno?
I don't have all your settings, autorizarion and an exemple of the file you try to send. Also, what command are you using? Did you try using the one file sharing option?

**Reply by dev-hanzo** [reply]
I changed the name of the repo. 
The plugin removed my MD files from the previous repo. 
It didn't delete the png files. 
It didn't send the files to the new repo

**Reply by dev-hanzo** [reply]
tomorrow I will try to reinstall the plugin and start from the beginning

**Reply by dev-hanzo** [reply]
thank you, we are close to success!

**Comment by dev-hanzo** [comment]
I did try made same whot made yesterday, but plugin not work again.
 I think the problem at possibility change repo name. 
**Plugin did not remembers new repos name**. 😟



![image](https://user-images.githubusercontent.com/117753674/210742214-3b8b3d98-f313-43d7-85ca-3a2762438fcb.png)

**Reply by Mara-Li** [reply]
That's pretty strange. I don't have the problem on my repo... 
Normally, settings must be saved, because when you open/close the settings, the settings are loaded properly.

Yesterday, I tested all commands and I don't get any error like that. 

Could you, please, check your `data.json` file, and check the repository?

Also, could you, please, try to send a file like that, using the command `share one file` : 


```md
---
dg-publish: true
title: Somethings testing
---

contents 
```

Maybe one of your file is broken... But it's strange it broke the repository checking...

**Reply by dev-hanzo** [reply]
![image](https://user-images.githubusercontent.com/117753674/210748165-77aba683-b2bf-4e5a-8688-b8b8a9db966a.png)

**file not sent in repo**

**Reply by Mara-Li** [reply]
At this point, I think your Github Publisher is haunted by my past ghost (joking).

I think the branch for the adding is not properly created. Or the main branch is not checked, for some reason my function doesn't stop when it check it. 

[...]

Okay, I know why it doesn't stop at this point. 

Could you, please, check if the branch (named by your vault-name) is created on the target repo ? 
Also, what is your vault name ?

**Reply by dev-hanzo** [reply]
I seem to have succeeded! one moment...

**Comment by dev-hanzo** [comment, marked as answer]
Wonderful, this story went the end!✨⭐

I think understood the problem.
**The problem was in name repo**. 
Decision - ⚡**reload obsidian**⚡ after change name repo.
So, that is what happened.

1. The plugin really not worked by me before v. 4.11.1
2. I setup v. 4.11.1 and I managed one time I sent my files
3. Then I changed name repo and everything is broken
4. **Plugin wont not updated name repo after changed settings. It need reload Obsidian!**
5. Now all right, I can sent my files to any my repo!

_I have one question._
If we use netlify then we can use private repo. Right?
But plugin can not work whit private repo. 
You can add this function?

**Reply by Mara-Li** [reply]
Well, it works on my side with a private repo

**Reply by dev-hanzo** [reply]
wow! I see now! It really work with private repo! 🔥
Thank you very much for your help and this plugin! 🙏

**Comment by dev-hanzo** [comment]
I see problem in file requirements.txt
it is error by Netlify

```
8:56:05 AM: ERROR: Could not find a version that satisfies the requirement obsidiantools==0.9.0 (from -r requirements.txt (line 15)) (from versions: 0.5.0, 0.6.0, 0.7.0, 0.8.0, 0.8.1)
8:56:05 AM: ERROR: No matching distribution found for obsidiantools==0.9.0 (from -r requirements.txt (line 15))
```

**Reply by Mara-Li** [reply]
Yeah, netlify only support python 3.8. You need to set obsidian tool to 0.8.2

**Comment by dev-hanzo** [comment]
Hi, I'm fine, thanks! can the plugin use gitlab instead of github? If not, do you plan to make this feature?

**Reply by Mara-Li** [reply]
This question was already replied multiple time.

No and no.

Please read the doc before askip question, and open new thread for new question. It will help people to find informations they need
